solarforecastarbiter/metrics/calculator.py

```diff
@@ -154,6 +157,11 @@ def calculate_deterministic_metrics(processed_fx_obs, categories, metrics,
                     'observation': obs,
                     'reference': ref_fx}, axis=1)
 
+    # Force `groupby` to be consistent with `interval_label`, i.e., if
+    # `interval_label == ending`, then the last interval should be in the bin
+    if processed_fx_obs.interval_label == "ending":
+        df.index -= pd.Timedelta("1ns")
```

Comment:
can you add a specific test for the proper behavior depending on interval length? I.e. run ``calculate_deterministic_metrics`` on one processed_fx_obs with the various categories for a one simple metric like MBE. Then compare the value to what you would expect e.g. for category == 'hour'
```python
index = pd.DatetimeIndex(['20191210T1300Z', '20191210T1330Z', '20191210T1400Z'])
interval_length = '1h'
fx_vals = pd.Series([0, 1, 2], index=index)
obs_vals = pd.Series([1, 1, 1], index=index)
...
res = calculator.calculate_deterministic_metrics(proc_fx_obs, ['hour'], ['mbe'])
# for 'ending'
assert {(v.index, v.value) for v in res.values} == {(12, -1.0), (13, 0.5)}
# for beginning
assert {(v.index, v.value) for v in res.values} == {(13, -0.5), (14, 1.0)}
```

Reply:
Sure, I'll work on adding that kind of test.

Reply:
@alorenzo175 I just added a specific test for the proper behavior depending on the interval label (see commit 8c2fa03). It's not a very pretty test, but it works well and covers the main cases.